### DISCUSSION TITLE
Getting CORS issue

### DISCUSSION THREAD

**Discussion by adityaNirvana** [dogecoin/dogecoin#3542] [Help needed]
Using version:
Dogecoin Core 1.14.7
Latest
on Feb 28

I can not find any option for CORS configuration in file https://github.com/dogecoin/dogecoin/blob/master/contrib/debian/examples/dogecoin.conf

Getting error:

```
Request has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
```

for method:

```
fetch('http://<IP>, {
            method: 'POST',
            headers: {
                'Authorization': 'Basic <auth-token>',
                'Content-Type': 'application/json'
            },
            body: JSON.stringify({
                "jsonrpc": "1.0",
                "id": "1",
                "method": "getblockcount",
                "params": []
            })
        })

Could anyone please help with correct rpc cors flag?

**Comment by patricklodder** [comment]
We have not implemented CORS simply because exposing the RPC to browsers is strongly discouraged, as that exposes administrative methods too.

Instead, use the REST interface if you really need to expose the daemon to browsers - you can enable it with `rest=1` in your `dogecoin.conf` file, [docs here](https://github.com/dogecoin/dogecoin/blob/a6d12201363b0eb4ccf854b609fc6e98958b1443/doc/REST-interface.md). If something is missing from this interface for you, please let us know by [filing a feature request](https://github.com/dogecoin/dogecoin/issues/new?assignees=&labels=&projects=&template=feature_request.md&title=%5Bfeat%5D+).

**Comment by adityaNirvana** [comment]
@patricklodder , Thanks for the reply! Actually getting same issue with rest endpoint too.

```
fetch('http://<IP>/rest/blockhashbyheight/5224895.json', {
            method: 'GET',
            headers: {
                'Authorization': 'Basic <auth-token>',
                'Content-Type': 'application/json'
            },
        })
```

my conf file:

```
server=1
rpcuser=*****
rpcpassword=*******
rpcport=22555
rpcallowip=0.0.0.0/0
setbackupdir=/datadir/backup
datadir=/datadir
rest=1
```

Error:

```
http://<IP>/rest/blockhashbyheight/5224895.json' from origin 'null' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
```

**Reply by patricklodder** [reply]
Do you expose this data to the internet or just locally?

**Comment by adityaNirvana** [comment]
@patricklodder, We have to expose it to internet & we are using our own authentication and authorization layer for that.
